src/cogs/PrequisiteChecker.py

```diff
@@ -13,8 +15,40 @@ def __init__(self, client):
         self.client = client
 
     @commands.command()
-    async def ping(self, ctx):
-        await ctx.send("Pong!")
+    async def prereq(self, ctx, arg):
+
+        # A lot of error checking and log messages
+        if len(arg) != 7:
+            await ctx.send("Invalid Input")
+            return
+
+        program = arg[0:4]
+        if program != "cpen" and program != "elec":
+            await ctx.send("Unable to identify specified program")
+            return
+
+        courseNumString = arg[4:7]
+        try:
+            int(courseNumString)
+        except: 
+            await ctx.send("Invalid Course Number")
+            return          
+
+        # Parent directory of the bot repo; constructed as parentDir(srcDir(fileDir(file)))
+        botDir = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+        course_info_file = open(os.path.join(botDir, 'assets/ece-course-prereqs.csv'), mode='r')
+
+        # Store course info csv as a collection of dictionaries
+        csv_reader = csv.DictReader(course_info_file, delimiter=',')
```

Comment:
I was wondering, is there a better way to do this? Every single time, we have to read through the entire dictionary until a matching course is found.
Would it be possible to pre-populate the dictionary on initialization? Then, we can do a check such as `if arg in dictionary:` instead of iterating through each row.